std/v2alpha/range/primitives.d

```diff
@@ -0,0 +1,33 @@
+module std.v2alpha.range.primitives;
+
+/**
+@@@TODO@@@ Publicly imported names across modules should copy or link those
+names' respective documentation. Here, we reuse empty and isInputRange because
+we don't change their meaning in v2.
+*/
+public import std.range.primitives : empty, isInputRange;
+
+/**
+@@@TODO@@@ This function redefines `front` for v2, meaning its documentation
+will override the documentation of `front` found in v1. The difference is of
+course that the new `front` does not autodecode.
+*/
+@property ref inout(T) front(T)(return scope inout(T)[] a) @safe pure nothrow @nogc
+if (!is(T[] == void[]))
+{
+    assert(a.length, "Attempting to fetch the front of an empty array of " ~ T.stringof);
+    return a[0];
+}
+
+/**
+@@@TODO@@@ This function redefines `popFront` for v2, meaning its documentation
+will override the documentation of `popFront` found in v1. The difference is of
+course that the new `popFront` does not autodecode.
+*/
+void popFront(T)(scope ref inout(T)[] a) @safe pure nothrow @nogc
+if (!is(T[] == void[]))
+{
+    assert(a.length, "Attempting to popFront() past the end of an array of " ~ T.stringof);
+    a = a[1 .. $];
+}
```

Comment:
what about

```D
private deprecated void popFrontString(scope ref string a) {
    // auto decode impl
}

void popFront(T)(scope ref inout(T)[] a) {
    static assert(!is(T[] == void[]), T.stringof ~ "[] not a support range");
    static if(is(T == string)) {
        popFrontString(a);
    } else {
        a = a[1 .. $];
    }
}
```

No need for std2, you should get a message ... move on

Reply:
No, we don't want to remove strings as ranges.

Reply:
after the deprecation phase we would remove everything but the first static assert and the a = a[1 .. $];

strings should still work, only without autodecoding.

What did I miss?